guide/src/tutorials/local-chains/start-relaying.md

```diff
@@ -87,7 +87,7 @@ Now, let's exchange `samoleans` between two chains.
 
 - Transfer back these tokens to ibc-0:
     ```shell
-    {{#template ../../templates/commands/hermes/transfer_with_denom dst-chain=ibc-1 src-chain=ibc-0 src-port=transfer src-channel=channel-0 amount=100000 timeout-seconds=10000 --denom ibc/C1840BD16FCFA8F421DAA0DAAB08B9C323FC7685D0D7951DC37B3F9ECB08A199}}
+    {{#template ../../templates/commands/hermes/tx/ft-transfer.md DST_CHAIN_ID=ibc-1 SRC_CHAIN_ID=ibc-0 SRC_PORT_ID=transfer SRC_CHANNEL_ID=channel-0 AMOUNT=100000 OPTIONS=--timeout-seconds=10000 --denom ibc/C1840BD16FCFA8F421DAA0DAAB08B9C323FC7685D0D7951DC37B3F9ECB08A199}}
```

Comment:
@seanchen1991 There are many cases in the guide where some commands need optional flags and I have to rely on "OPTIONS=". 

If we rename these flags then the command will not work. I am not sure how to deal with this.

Reply:
Exactly. I can't think of any solution that is 100% automatic. A possible solution would be to detect if an optional flag has changed for a given command and to tell the user to update the guide accordingly.

Reply:
Yea, I was thinking that's probably the best way to go about this as well

Reply:
The guide can also break if we modify the name of a command. For example, if instead of `hermes tx subcommand` we had `hermes tx raw subcommand`.

Reply:
Yea, I think the best thing to do in either of those cases would be for the CI job that generates the template files to fail, indicating to the PR author that the commands need to be updated in the guide.

Reply:
I looked at the source code of `mdbook-template`.

I think it's possible to retrieve all the pieces of code that use a command template in the guide, inject the dynamic arguments ([example](https://github.com/sgoudham/mdbook-template/blob/5da795954563a401b29bf0ac9dc4e5fbf9d5678e/src/lib.rs#L173)) and verify that the command can be parsed with [`try_parse_from()`](https://docs.rs/clap/3.0.14/clap/trait.Parser.html#method.parse_from). 

It won't work with new commands but it will work with current ones.

Reply:
Nevermind, the function `replace_args` is not public. I can still make my own.

Reply:
> I think it's possible to retrieve all the pieces of code that use a command template in the guide, inject the dynamic arguments ([example](https://github.com/sgoudham/mdbook-template/blob/5da795954563a401b29bf0ac9dc4e5fbf9d5678e/src/lib.rs#L173)) and verify that the command can be parsed with [try_parse_from()](https://docs.rs/clap/3.0.14/clap/trait.Parser.html#method.parse_from).

I'm not clear this will change the usage flow from what we previously mentioned.

Reply:
> > I think it's possible to retrieve all the pieces of code that use a command template in the guide, inject the dynamic arguments ([example](https://github.com/sgoudham/mdbook-template/blob/5da795954563a401b29bf0ac9dc4e5fbf9d5678e/src/lib.rs#L173)) and verify that the command can be parsed with [try_parse_from()](https://docs.rs/clap/3.0.14/clap/trait.Parser.html#method.parse_from).
> 
> I'm not clear this will change the usage flow from what we previously mentioned.

It's true however I think it could be an improvement. With the current usage flow, the script will just detect a change 
without giving details about what changed and what needs to be updated. It will be the responsability of the user to update the guide and make sure every command is syntaxically correct.

Another solution is to verify that every command can be parsed. It's a bit more work but the script will throw errors until every command can be parsed.

Reply:
Maybe we can leave it as a suggestion for now and eventually come back to it later.